Extension/src/LanguageServer/client.ts

```diff
@@ -1135,10 +1135,9 @@ export class DefaultClient implements Client {
                     return ui.ShowConfigureIntelliSenseButton(false, this, ConfigurationType.CompileCommands, showButtonSender);
                 } else {
                     action = "select compiler";
-                    const newCompiler: string = util.isCl(paths[index]) ? "cl.exe" : paths[index];
-
+                    let newCompiler: string = util.isCl(paths[index]) ? "cl.exe" : paths[index];
+                    newCompiler = newCompiler.replace(/[\\/]/g, preferredPathSeparator);
```

Comment:
FYI: `path.normalize( )` is the node built-in function to ensure paths are formatted with the correct path separators.